itou/www/dashboard/forms.py

```diff
@@ -305,40 +304,67 @@ def clean_email(self):
         else:
-            NewSpontaneousJobAppEmployersNotification.unsubscribe(recipient=self.recipient)
+            disabled_notifications = []
+
+        previous_category = None
+        for notification_class in notification_registry:
+            notification = notification_class(self.user, self.structure)
+            notification_class_name = notification_class.__name__
+            category_slug = slugify(notification.category)
+
+            if notification.is_manageable_by_user():
+                if previous_category is None or previous_category != notification.category:
+                    previous_category = notification.category
+                    self.fields[f"category-{category_slug}-all"] = forms.BooleanField(
+                        required=False,
+                        label="Toutes les notifications",
+                        initial=False if notification_class_name in disabled_notifications else True,
+                        widget=forms.CheckboxInput(
+                            attrs={
+                                "class": f"category-{category_slug} category-grouper",
+                                "data-category-name": notification.category,
+                                "data-category-slug": category_slug,
+                            }
+                        ),
+                    )
+                    self.layout[category_slug] = {"name": notification.category, "notifications": []}
+
+                self.fields[notification_class_name] = forms.BooleanField(
+                    required=False,
+                    label=notification.name,
+                    initial=False if notification_class_name in disabled_notifications else True,
+                    widget=forms.CheckboxInput(
+                        attrs={
+                            "class": f"category-{category_slug}",
+                            "data-category-slug": category_slug,
+                        }
+                    ),
+                )
+                self.layout[category_slug]["notifications"].append(notification_class_name)
 
-        if self.company.job_description_through.exists():
-            to_subscribe_pks = self.cleaned_data.get("qualified")
-            NewQualifiedJobAppEmployersNotification.replace_subscriptions(
-                recipient=self.recipient, subscribed_pks=to_subscribe_pks
-            )
+    def save(self):
+        notification_settings = NotificationSettings.get_or_create(self.user, self.structure)
+        disabled_notifications = []
+
+        for field_name, value in self.cleaned_data.items():
+            if field_name.startswith("category-"):
+                continue
+            if not value:
+                notification = NotificationRecord.objects.get(notification_class=field_name)
+                if not self.cleaned_data[f"category-{slugify(notification.category)}-all"]:
+                    disabled_notifications.append(notification)
```

Comment:
C’est étrange de vérifier ces composants, qui ne servent qu’au display. Pourquoi ne pas suivre la valeur de la checkbox ?

```suggestion
                disabled_notifications.append(notification)
```

Note : ça devrait aussi aider à réduire le nombre de requêtes SQL.